Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim-buster as base
+# Install poetry
+RUN pip install poetry
+WORKDIR /opt
+# Expose the port that we want the host to listen to
+EXPOSE 3000
+# Copy across the project files
+COPY . /opt
+
+FROM base as development
+ENV FLASK_ENV=development
+# Install the project dependencies
+RUN poetry install
+# Set the script to run as development on startup of the container
+ENTRYPOINT ["poetry", "run", "flask", "run", "--host=0.0.0.0", "--port=8000"]
+
+FROM base as production
+# Install the project dependencies
+RUN poetry install
```

Comment:
Given that this happens in both stages, we could move it into the `base` stage (i.e. below the `COPY . /opt` so we only need it once, but it's very minor (and your approach actually allows us to later introduce differences, such as skipping certain dependencies in some environments)